contrib/endpoints/src/api_manager/config_manager.cc

```diff
@@ -106,17 +93,52 @@ void ConfigManager::OnFetchAuthTokenDone(utils::Status status) {
   if (!status.ok()) {
     // We should not get here
     global_context_->env()->LogError("Unexpected status: " + status.ToString());
-    rollout_apply_function_(utils::Status(Code::ABORTED, status.ToString()),
-                            kEmptyConfigs);
     return;
   }
 
-  // Fetch configs from the Inception
-  // For now, config manager has only one config_id (100% rollout)
-  std::shared_ptr<ConfigsFetchInfo> config_fetch_info(
-      new ConfigsFetchInfo({{global_context_->config_id(), 100}}));
-
-  FetchConfigs(config_fetch_info);
+  rollouts_refresh_timer_ = global_context_->env()->StartPeriodicTimer(
+      std::chrono::milliseconds(refresh_interval_ms_), [this]() {
+        service_management_fetch_->GetRollouts([this](
+            const utils::Status& status, std::string&& rollouts) {
+          if (!status.ok()) {
+            global_context_->env()->LogError(
+                std::string("Failed to download rollouts: ") +
+                status.ToString() + ", Response body: " + rollouts);
+            return;
+          }
+
+          ListServiceRolloutsResponse response;
+          if (!utils::JsonToProto(rollouts,
+                                  (::google::protobuf::Message*)&response)
+                   .ok()) {
+            global_context_->env()->LogError(std::string("Invalid response: ") +
+                                             status.ToString() +
+                                             ", Response body: " + rollouts);
+            return;
+          }
+
+          if (response.rollouts_size() == 0) {
+            global_context_->env()->LogError("No active rollouts");
+            return;
+          }
+
+          std::shared_ptr<ConfigsFetchInfo> config_fetch_info =
```

Comment:
here we need to compare rollout_id  with the existing new,   if same, do nothing.